src/draw/sw/lv_draw_sw_gradient.c

```diff
@@ -62,45 +69,44 @@ union void_cast {
     const uint32_t value;
 };
 
-static uint32_t compute_key(const lv_gradient_t * g, lv_coord_t size, lv_coord_t w)
+static uint32_t compute_key(const lv_grad_dsc_t * g, lv_coord_t size, lv_coord_t w)
 {
     union void_cast v;
     v.ptr = g;
     return (v.value ^ size ^ (w >> 1)); /*Yes, this is correct, it's like a hash that changes if the width changes*/
 }
 
-
-static size_t get_cache_item_size(lv_gradient_cache_t * c)
+static size_t get_cache_item_size(lv_grad_t * c)
 {
-    size_t s = sizeof(*c) + c->alloc_size * sizeof(lv_color_t)
+    size_t s = ALIGN(sizeof(*c)) + ALIGN(c->alloc_size * sizeof(lv_color_t));
 #if _DITHER_GRADIENT
-               + c->size * sizeof(lv_color32_t)
+    s += ALIGN(c->size * sizeof(lv_color32_t));
 #if LV_DITHER_ERROR_DIFFUSION == 1
-               + c->w * sizeof(lv_scolor24_t)
+    s += ALIGN(+ c->w * sizeof(lv_scolor24_t));
```

Comment:
Extra + in there ?

Reply:
I don't understand. How do you mean it?

Reply:
The leading + is redundant before `c->w`, I think.

Reply:
Ups, fixed. I'm a little bit blind sometimes :laughing: